### ISSUE TITLE
Questions about the experiments

### ISSUE THREAD

**Issue by Alex-Wula** [md4all/md4all#23]
Dear authors:
Thank you for your work again !  I want to ask some questions about the experiments.  
First, I would like to know when your team was training forkgan to generate images, did you encounter the problem that the quality of the generated images was low, and the distribution of generated images was greatly different with the nuscenes night scene images, which in turn led to the poor performance of md4all? 
Second, I'm wondering if your team's metrics for night shook up badly when you were training md4all? When I was training md4all using my own gan generated images, I found that the nightscape metrics oscillated badly and did not converge.  
I would be very grateful if you could reply to me. Thank you for reading my question in your busy schedule !

**Comment by morbi25** [comment]
Hi @Alex-Wula,

Thank you for your question!

1. As written in Section A.2.1 of the paper, the training of the ForkGAN on nuScenes can be indeed challenging. In our experiments, we observed that ForkGAN needs many images per condition (around 20000 - 30000) to converge well and produce reasonable translations. However, the nuScenes dataset only provides ~5000 night images. 
To be able to train ForkGAN anyway, you can try the following steps:
1.1 Complement nuScenes-night with the night images of nuImages as they seem to be from a similar data distribution. However, to do this you have to separate/label them manually as nuImages does not provide direct meta information about the current daytime or weather condition. Nevertheless, for night, you could potentially use the timestamp attribute to apply a coarse filtering.
1.2 Try to select only images for your conditions of interest that represent it well enough. More specifically, it can be beneficial to filter out images that belong to a boundary case of a particular condition. E.g. for the condition "night", you could filter out images that are recorded during the sunset but are labeled as "night".
1.3 If your translations are still not well enough, you can try to also pretrain ForkGAN on another dataset containing enough day and night images and then finetune it on nuScenes.
2. If the quality of your generated images is too low or their distribution is largely different from the nuScenes one, md4all's performance can degrade as it is evaluated on the real night images of the official nuScenes validation and test split and **not** on the translated images. So following the steps above (point 1) to improve the quality of the translated images could fix the issue of the oscillating and non-converging metrics for the condition "night".

Alternatively, you can also use our uploaded, translated images or our uploaded ForkGAN checkpoint. :)

I hope these steps help you! Have a nice week! :)

**Comment by Alex-Wula** [comment]
Dear @morbi25 
Thank you very much! Your answer is very helpful to me!
